compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -9343,4 +9375,342 @@ private STNode parseIntersectionTypeDescriptor(STNode leftTypeDesc, ParserRuleCo
+        } else {
+            return STNodeFactory.createTypedBindingPatternNode(typeDesc,
+                    parseBindingPattern());
+        }
+
+
+    }
+
+    private STNode mergeTypeDescAndListBindingPattern(STNode typeDesc,
+                                                      STNode bindingPattern) {
+        STListBindingPatternNode listBindingPattern = (STListBindingPatternNode) bindingPattern;
+        STNodeList childArray = (STNodeList) listBindingPattern.bindingPatterns;
+        STNode child = childArray.childInBucket(0);
+        if (child.kind == SyntaxKind.CAPTURE_BINDING_PATTERN) {
+            child = ((STCaptureBindingPatternNode) child).variableName;
+        }
+        return STNodeFactory.createIndexedExpressionNode(typeDesc, listBindingPattern.openBracket,
+                child,
+                listBindingPattern.closeBracket);
+    }
+
+    private STNode cleanListBindingPattern(STNode bindingPattern) {
+        STListBindingPatternNode listBindingPattern = (STListBindingPatternNode) bindingPattern;
+        STNodeList childArray = (STNodeList) listBindingPattern.bindingPatterns;
+        int numberOfChildren = childArray.bucketCount();
+
+        ArrayList<STNode> cleanedChildren = new ArrayList<>();
+        STNode child = null;
+        for (int i = 0; i < numberOfChildren; i++) {
+            child = childArray.childInBucket(i);
+            switch (child.kind) {
+                case DECIMAL_INTEGER_LITERAL:
+                case HEX_INTEGER_LITERAL:
+                case ASTERISK_TOKEN:
+                    this.errorHandler.reportInvalidNode(null,
+                            "invalid Node " + child.kind + " in list-binding-pattern");
+                    cleanedChildren.add(STNodeFactory.createMissingToken(SyntaxKind.IDENTIFIER_TOKEN));
+                    break;
+                default:
+                    cleanedChildren.add(child);
+                    break;
+            }
+        }
+
+        return STNodeFactory.createListBindingPatternNode(listBindingPattern.openBracket,
+                STNodeFactory.createNodeList(cleanedChildren),
+                listBindingPattern.restBindingPattern,
+                listBindingPattern.closeBracket);
+    }
+
+    private boolean isListBindingPatternDefinitively(STNode bindingPattern) {
+        STListBindingPatternNode listBindingPattern = (STListBindingPatternNode) bindingPattern;
+        STNodeList childArray = (STNodeList) listBindingPattern.bindingPatterns;
+        int numberOfChildren = childArray.bucketCount();
+
+        if (listBindingPattern.restBindingPattern != null) {
+            return true;
+        }
+
+        if (numberOfChildren == 1) {
+            STNode child = childArray.childInBucket(0);
+            switch (child.kind) {
+                case LIST_BINDING_PATTERN:
+                case REST_BINDING_PATTERN:
+                    return true;
+                default: // cases like [ and {
+                    return false;
```

Comment:
I think its safe to do the reverse of this. 
i.e: if the only-child is captured-binding-patterns, then return true.
That way we don't have to update this method, when we add new binding patterns.

Reply:
sure